### PR TITLE
IDVA6-1192: Bookmark Page Redirects

### PR DESCRIPTION
**Pull request by kwamenyantakyi-ch**
**Changes made:**
- Added navigationMiddleware
- Implemented bookmark page redirects
- Added unit tests for navigationMiddleware
- Added Navigation Types

Link to ticket: https://companieshouse.atlassian.net/browse/IDVA6-1192

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/middleware/navigationMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+export const NAVIGATION: Navigation = {
+    [constants.CHECK_MEMBER_DETAILS_FULL_URL]: {
+        allowedReferers: [constants.ADD_USER_FULL_URL, constants.CHECK_MEMBER_DETAILS_FULL_URL, constants.CANNOT_ADD_USER_FULL_URL],
+        redirectTo: constants.MANAGE_USER_FULL_URL
+    },
+    [constants.CONFIRMATION_MEMBER_ADDED_FULL_URL]: {
+        allowedReferers: [constants.CHECK_MEMBER_DETAILS_FULL_URL, constants.CONFIRMATION_MEMBER_ADDED_FULL_URL],
+        redirectTo: constants.MANAGE_USER_FULL_URL
+    },
+    [constants.getRemoveMemberCheckDetailsFullUrl("")]: {
```
Comment (mmodzelewski-ch) [inline src/middleware/navigationMiddleware.ts:16]: There should be no functions in the constants file. Please move getRemoveMemberCheckDetailsFullUrl() function to utils.

**Review by mmodzelewski-ch — CHANGES_REQUESTED** [review]

File: src/middleware/navigationMiddleware.ts (hunk 1 above)
Comment (kwamenyantakyi-ch) [reply src/middleware/navigationMiddleware.ts]: Hi Maciej, I have resolved it, please double check again.

File: src/lib/utils/urlUtils.ts [hunk 2]
```diff
@@ -5,3 +5,5 @@ const WHITELISTED_URLS: string[] = [
 ];
 
 export const isWhitelistedUrl = (url: string): boolean => WHITELISTED_URLS.includes(url);
+
+export const getRemoveMemberCheckDetailsFullUrl = (id: string): string => `${constants.LANDING_URL}/${constants.REMOVE_MEMBER_PAGE}/${id}`;
```
Comment (mmodzelewski-ch) [inline src/lib/utils/urlUtils.ts:9]: Please add unit tests in an appropriate place to test this function.

**Review by mmodzelewski-ch — CHANGES_REQUESTED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/lib/utils/urlUtils.ts (hunk 2 above)
Comment (kwamenyantakyi-ch) [reply src/lib/utils/urlUtils.ts:9]: I have added unit test for this function. Please re-review.

**Review by mmodzelewski-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**